### PR TITLE
Extract API functions from admin-ui package

### PR DESCRIPTION
**Pull request by mattgrill**
## Issue
https://github.com/jsdrupal/drupal-admin-ui/issues/571

The most basic set of helper functions extracted from the `admin-ui` package.

### CODE REVIEW

File: packages/utilities/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# @drupal/admin-ui-components
+
+Helper utilities for Admin UI and extension point projects.
+
+## `api`
+
+An `async` helper function for making requests to a Drupal backend.
+
+- `@param {string} REACT_APP_DRUPAL_BASE_URL` The base url of the backend (Drupal)
```
Comment (dawehner) [inline packages/utilities/README.md]: Do you mind copying this documentation to the api function itself, so tools could show it?

File: packages/admin-ui/src/utils/api/api.js [hunk 2]
```diff
@@ -1,217 +1,5 @@
...
-
-    // CSRF tokens return text, not json.
-    if (options.text) {
-      return res.text();
-    }
-    return res.json();
-  });
-  return data;
-}
+const api = apiHelper.bind(null, process.env.REACT_APP_DRUPAL_BASE_URL);
```
Comment (dawehner) [inline packages/admin-ui/src/utils/api/api.js]: It works, its not nice, but hey, its progress!

File: packages/utilities/src/api.js [hunk 3]
```diff
@@ -0,0 +1,236 @@
+import qs from 'qs';
+import { ApiError } from './errors';
+
+/**
+ * An async helper function for making requests to a Drupal backend.
+ *
+ * @param {string} REACT_APP_DRUPAL_BASE_URL
```
Comment (dawehner) [inline packages/utilities/src/api.js]: Nice, thank you!

Comment (mattgrill) [reply packages/utilities/src/api.js]: 🙏

File: packages/admin-ui/src/utils/api/api.js (hunk 2 above)
Comment (mattgrill) [reply packages/admin-ui/src/utils/api/api.js]: I think this small change might resolve not nice'ness?

File: packages/utilities/package.json [hunk 4]
```diff
@@ -0,0 +1,26 @@
+{
+  "name": "@drupal/admin-ui-utilities",
```
Comment (lauriii) [inline packages/utilities/package.json]: What other features are we planning to include in this package?

Comment (mattgrill) [reply packages/utilities/package.json]: @lauriii There was some discussion in #571 about this. I don't think for now anything outside the api utilities are needed.

Comment (dawehner) [reply packages/utilities/package.json]: For me this is ready to be merged.

**Pull request merged**